tests/Endatix.Api.Tests/Endpoints/Submissions/GetByIdTests.cs

```diff
@@ -0,0 +1,111 @@
+using FastEndpoints;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Endatix.Core.Infrastructure.Result;
+using Endatix.Core.Entities;
+using Endatix.Api.Endpoints.Submissions;
+using Endatix.Core.UseCases.Submissions.GetById;
+
+namespace Endatix.Api.Tests.Endpoints.Submissions;
+
+public class GetByIdTests
+{
+    private readonly IMediator _mediator;
+    private readonly GetById _endpoint;
+
+    public GetByIdTests()
+    {
+        _mediator = Substitute.For<IMediator>();
+        _endpoint = Factory.Create<GetById>(_mediator);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_InvalidRequest_ReturnsBadRequest()
+    {
+        // Arrange
+        var formId = 1L;
+        var submissionId = 1L;
+        var request = new GetByIdRequest { FormId = formId, SubmissionId = submissionId };
+        var result = Result.Invalid();
+        
+        _mediator.Send(Arg.Any<GetByIdQuery>(), Arg.Any<CancellationToken>())
+            .Returns(result);
+
+        // Act
+        var response = await _endpoint.ExecuteAsync(request, default);
+
+        // Assert
+        var badRequestResult = response.Result as BadRequest;
+        badRequestResult.Should().NotBeNull();
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_SubmissionNotFound_ReturnsNotFound()
+    {
+        // Arrange
+        var formId = 1L;
+        var submissionId = 1L;
+        var request = new GetByIdRequest { FormId = formId, SubmissionId = submissionId };
+        var result = Result.NotFound("Submission not found");
+
+        _mediator.Send(Arg.Any<GetByIdQuery>(), Arg.Any<CancellationToken>())
+            .Returns(result);
+
+        // Act
+        var response = await _endpoint.ExecuteAsync(request, default);
+
+        // Assert
+        var notFoundResult = response.Result as NotFound;
+        notFoundResult.Should().NotBeNull();
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ValidRequest_ReturnsOkWithSubmission()
+    {
+        // Arrange
+        var formId = 1L;
+        var submissionId = 1L;
+        var request = new GetByIdRequest { FormId = formId, SubmissionId = submissionId };
+        var submission = new Submission("{ }", 1, 2) { Id = submissionId };
```

Comment:
Sharing thoughts.

We need some domain testing infrastructure - set of test Ids, set of test JSON definitions and responses for quick db or test seeding. 

We should try to make tests easier to write and read by humans so less things to worry about the better

Ideally, this should be shared across all projects in the test folder

Reply:
Generally, this is considered good practice but before going in this direction, we need to consider how it will affect the AI assistance we use for tests generation. I don't know if this is the case, but it is possible that generating/writing tests could become more time consuming if we need to use seed data.

Reply:
It shouldn't matter. It should help us actually. AI will predict the use of sample data and factory/data seeding methods once we start using the. Copilot and Cursor index the codebase, so Thai will help both humans and agents